src/schema/sources.ts

```diff
@@ -975,6 +990,12 @@ const validateSquadData = async (
     throw new ValidationError('Invalid member invite role');
   }
 
+  if (moderationRequired && memberPostingRole !== SourceMemberRoles.Member) {
```

Comment:
One edge case which comes to mind, if client does not send `memberPostingRole` it will act as if moderator only because `undefined !== SourceMemberRoles.Member`. Maybe we should check the actual value set on squad in the db if memberPostingRole is missing in mutation?

On the other hand I don't see why this would even be a requirement on API, to simplify we can just accept moderationRequired always, or is it something I am missing?

Reply:
@capJavert I mean you're not entirely wrong that it wouldn't really matter, as we are going to check if the user is admin and if moderation is required when posts are created to determine if we need to create a SourcePostModeration entry or not. Having the validation on frontend only should suffice, as the purpose of the validation is purely an UX one.

Want me to remove the check?

Reply:
Yeah, I think it will be easier if we just set it with whatever that user passes, as long as the user has the right user privilege for the Squad.

Reply:
https://github.com/dailydotdev/daily-api/pull/2324#discussion_r1798963928